cookbook/cache/form_csrf_caching.rst

```diff
@@ -0,0 +1,54 @@
+.. index::
+    single: Cache; CSRF; Forms
+
+Caching Pages that Contain CSRF Protected Forms
+===============================================
+
+CSRF - or `Cross-site request forgery`_ - is a method by which a malicious
+user attempts to make your legitimate users unknowingly submit data that
+they don't intend to submit. Fortunately, CSRF attacks can be prevented by
+using a CSRF token inside your forms.
+
+For more information about how this protection works in Symfony, please
+check :ref:`CSRF Protection <forms-csrf>`.
+
+.. note::
+
+    A `Security CSRF Component`_ was introduced in Symfony 2.4 which provides
+    a class CsrfTokenManager for generating and validating CSRF tokens
+
+Why Varnish does not cache by default these pages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are many ways to generate unique tokens for each user but in order get
+them validated when the form is submitted, you need to store them inside the
+PHP Session.
+
+If you are using Varnish or some similar reverse proxy cache and you try to cache
+pages containing forms with CSRF token protection, you will see that, by default,
+Varnish fails to cache them.
```

Comment:
The `HttpCache` class absolutely *does* cache responses with cookies. @ricardclau said otherwise, but in his to the code (https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php#L675:L690), you'll notice that `isPrivateRequest()` is only used here: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php#L508. In other words, if the response is public, then it *will* be cached by Symfony's reverse proxy.

As I understand it, that *is* different than Varnish, where you need to do some VCL stuff, because by default it will not cache any response with headers. Is that right?

My point is that this should be re-worded a little bit: some reverse proxies refuse to cache responses with cookies by default (Varnish) while others will cache them, which will mean that every user after the first will get a response with the wrong CSRF token.